File: movie-db/src/api/sql_user.ts
interface SqlUser {
    username: string,
    is_admin: boolean,
    created_at: string,
}


interface SqlUserProps {
    username: string;
    password: string;
    is_admin: boolean;
}

type SqlUserList = SqlUser[];
export {
    type SqlUser,
    type SqlUserProps,
    type SqlUserList,
}

File: movie-db/src-tauri/src/sql/types/sql_user.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct SqlUserCredentials {
    pub username: String,
    pub password: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SqlUser {
    pub username: String,
    pub is_admin: bool,
    pub created_at: String,
}

impl SqlUser {
    pub fn from_row(row: &oracle::Row) -> Result<Self, oracle::Error> {
        Ok(Self {
            username: row.get(0)?,
            is_admin: match row.get(1)? {
                1 => true,
                _ => false,
            },
            created_at: row.get(2)?,
        })
    }
}